### PR TITLE
Add support for Python 3.11.0a5

### LINKED ISSUE #28: Does not build on Python 3.11.0a3
### What I did:

``tox -epy311``

### What I expect to happen:

Clean build without any errors.

### What actually happened:

```
$ tox -epy311
py311 develop-inst-nodeps: /.../Persistence
ERROR: invocation failed (exit code 1), logfile: /.../Persistence/.tox/py311/log/py311-4.log
======================================================================== log start =========================================================================
Looking in indexes: https://pypi.org/simple, https://mihowitz:****@development.verdi.de/devpi/verdi/prod/+simple/
Obtaining file:///.../Persistence
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Installing collected packages: Persistence
  Running setup.py develop for Persistence
    ERROR: Command errored out with exit status 1:
     command: /.../Persistence/.tox/py311/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/.../Persistence/setup.py'"'"'; __file__='"'"'/.../Persistence/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: /.../Persistence/
    Complete output (22 lines):
    running develop
    /.../Persistence/.tox/py311/lib/python3.11/site-packages/setuptools/command/easy_install.py:156: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    /.../Persistence/.tox/py311/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    running egg_info
    writing src/Persistence.egg-info/PKG-INFO
    writing dependency_links to src/Persistence.egg-info/dependency_links.txt
    writing requirements to src/Persistence.egg-info/requires.txt
    writing top-level names to src/Persistence.egg-info/top_level.txt
    reading manifest file 'src/Persistence.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    adding license file 'LICENSE.txt'
    writing manifest file 'src/Persistence.egg-info/SOURCES.txt'
    running build_ext
    building 'Persistence._Persistence' extension
    /usr/bin/clang -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk -Iinclude -Isrc -I/.../Persistence/.tox/py311/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c src/Persistence/_Persistence.c -o build/temp.macosx-11.0-x86_64-3.11/src/Persistence/_Persistence.o
    src/Persistence/_Persistence.c:178:19: error: expression is not assignable
      Py_TYPE(&Ptype) = ECExtensionClassType;
      ~~~~~~~~~~~~~~~ ^
    1 error generated.
    error: command '/usr/bin/clang' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /.../Persistence/.tox/py311/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/.../Persistence/setup.py'"'"'; __file__='"'"'/.../Persistence/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.

========================================================================= log end ==========================================================================
_________________________________________________________________________ summary __________________________________________________________________________
ERROR:   py311: InvocationError for command /.../Persistence/.tox/py311/bin/python -m pip install --no-deps -e '/.../Persistence[test]' (exited with code 1)
```

### What version of Python and Zope/Addons I am using:

* `master` of `Persistence`
* Python 3.11.0a3
* MacOS Big Sur

### PR DESCRIPTION
**Pull request by icemac**
Fixes #28.

Open issues:

- [x] Installation on Python 3.11 fails during compilation, see #28 and https://github.com/zopefoundation/ExtensionClass/pull/38 for possible fixes.

### CODE REVIEW

**Comment by icemac** [comment]
@dataflake I think so, too. It is only a matter of having the time necessary 😁.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -167,7 +170,7 @@ jobs:
         # We cannot 'uses: pypa/gh-action-pypi-publish@v1.4.1' because
         # that's apparently a container action, and those don't run on
         # the Mac.
-        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags') && startsWith(runner.os, 'Mac')
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags') && startsWith(runner.os, 'Mac') && !startsWith(matrix.python-version, 'pypy')
```
Comment (d-maurer) [inline .github/workflows/tests.yml:173]: Could this line be shortened?

**Review by d-maurer — APPROVED** [review]

**Review by dataflake — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (icemac) [reply .github/workflows/tests.yml]: Done.

**Comment by icemac** [comment]
Updating to 3.11a4 breaks the ubuntu build but MacOS stays green.
See https://github.com/zopefoundation/Persistence/runs/4878745140?check_suite_focus=true

Locally on MacOS it is green, too.
Any ideas?

**Comment by icemac** [comment]
3.11a4 on MacOS just breaks inside coverage.

**Comment by dataflake** [comment]
I don't see any local failures on macOS with Python 3.11a4.

So this breaks on compiling the `persistent` package, which is a dependency. I'll note that according to its `setup.py` it only supports up to Python 3.9. It might help to go there first and ensure full compatibility before continuing here.

**Comment by icemac** [comment]
@dataflake This is my plan now, too. `persistent` will not be so easy to tackle because it first hast to be ported to `src` layout. But I am going to look into it.

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**